crates/macros/src/parsing.rs

```diff
@@ -114,12 +114,23 @@ impl MethodRename for syn::Ident {
 
 impl MethodRename for &str {
     fn rename_method(&self, rule: &RenameRule) -> String {
-        let s = self.to_string();
-
-        if MAGIC_METHOD.contains(&s.as_str()) {
-            s
-        } else {
-            self.rename(rule)
+        match rule {
+            RenameRule::None => self.to_string(),
+            _ => {
+                if MAGIC_METHOD.contains(self) {
+                    if self == &MAGIC_METHOD[12] {
+                        "__toString".to_string()
+                    } else if self == &MAGIC_METHOD[16] {
+                        "__debugInfo".to_string()
+                    } else if self == &MAGIC_METHOD[3] {
+                        "__callStatic".to_string()
```

Comment:
Don't like the index reference here. Changing the order of the items would change this as well without being apparent.

I think it would be better to either compare it against a static string or store the map in `MAGIC_METHOD`.

Reply:
I agree, i wanna ask your advice about this